modules/services/dunst.nix

```diff
@@ -85,75 +89,87 @@ in
-              "emotes"
-              "filesystem"
-              "intl"
-              "mimetypes"
-              "places"
-              "status"
-              "stock"
-            ];
-          in
-            concatStringsSep ":" (
-              concatMap (theme:
-                concatMap (basePath:
-                  map (category:
-                    "${basePath}/share/icons/${theme.name}/${theme.size}/${category}"
-                  ) categories
-                ) basePaths
-              ) themes
+    {
+      xdg.configFile."dunst/dunstrc".source = configFile;
+      xdg.dataFile."dbus-1/services/org.knopwob.dunst.service".source =
+        "${pkgs.dunst}/share/dbus-1/services/org.knopwob.dunst.service";
+
+      services.dunst.settings.global.icon_folders =
+        let
+          useCustomTheme =
+            cfg.iconTheme.package != hicolorTheme.package
+            || cfg.iconTheme.name != hicolorTheme.name
+            || cfg.iconTheme.size != hicolorTheme.size;
+
+          basePaths = [
+            "/run/current-system/sw"
+            "${config.home.homeDirectory}/.nix-profile"
+            cfg.iconTheme.package
+          ] ++ optional useCustomTheme hicolorTheme.package;
+
+          themes =
+            [
+              cfg.iconTheme
+            ] ++ optional useCustomTheme (
+              hicolorTheme // { size = cfg.iconTheme.size; }
             );
 
-        systemd.user.services.dunst = {
-          Unit = {
-            Description = "Dunst notification daemon";
-            After = [ "graphical-session-pre.target" ];
-            PartOf = [ "graphical-session.target" ];
-          };
+          categories = [
+            "actions"
+            "animations"
+            "apps"
+            "categories"
+            "devices"
+            "emblems"
+            "emotes"
+            "filesystem"
+            "intl"
+            "mimetypes"
+            "places"
+            "status"
+            "stock"
+          ];
+        in
+          concatStringsSep ":" (
+            concatMap (theme:
+              concatMap (basePath:
+                map (category:
+                  "${basePath}/share/icons/${theme.name}/${theme.size}/${category}"
+                ) categories
+              ) basePaths
+            ) themes
+          );
+
+      systemd.user.services.dunst = {
+        Unit = {
+          Description = "Dunst notification daemon";
+          After = [ "graphical-session-pre.target" ];
+          PartOf = [ "graphical-session.target" ];
+        };
 
-          Service = {
-            Type = "dbus";
-            BusName = "org.freedesktop.Notifications";
-            ExecStart = "${pkgs.dunst}/bin/dunst";
-          };
+        Service = {
+          Type = "dbus";
+          BusName = "org.freedesktop.Notifications";
+          ExecStart = "${pkgs.dunst}/bin/dunst";
         };
+      };
+      home.activation.checkDunst = dagEntryBefore [ "linkGeneration" ] ''
+        if ! cmp --quiet \
```

Comment:
This should only happen if the user actually manage the configuration file through this module.